### PR TITLE
Define and simulate DensityMatrixDisplay

### LINKED ISSUE #1382: Add displays to density matrix simulator
That we have to reuse code for the displays feels bad to me, I wonder if there are some better abstractions that could help here.

### PR DESCRIPTION
**Pull request by kevinsung**
Fixes #1382 .

Also exposes `cirq.testing.random_superposition` and moves it into `cirq.testing.lin_alg_utils`.

### CODE REVIEW

File: cirq/testing/random_superposition_test.py [hunk 1]
```diff
@@ -1,11 +0,0 @@
-import numpy as np
-import pytest
-
-from cirq.testing.random_superposition import random_superposition
-
-@pytest.mark.parametrize('dim',  range(1, 10))
-def test_random_superposition(dim):
```
[the remark this answers is not in the document]
Comment (kevinsung) [reply cirq/testing/random_superposition_test.py]: Oops! Restored.

File: cirq/ops/display.py [hunk 2]
```diff
@@ -98,9 +98,32 @@ def value_derived_from_wavefunction(self,
...
-            state: The wavefunction
+            state: The wavefunction.
             qubit_map: A dictionary from qubit to qubit index in the
-                ordering used to define the wavefunction
+                ordering used to define the wavefunction.
+        """
+        pass
+
+
+class DensityMatrixDisplay(raw_types.Operation):
```
[the remark this answers is not in the document]
Comment (kevinsung) [reply cirq/ops/display.py]: Done

File: cirq/testing/lin_alg_utils.py [hunk 3]
```diff
@@ -21,6 +21,19 @@
 from cirq import linalg
 
 
+def random_superposition(dim: int) -> np.ndarray:
+    """Returns a random unit-length vector from the uniform distribution.
```
Comment (Strilanc) [inline cirq/testing/lin_alg_utils.py]: Double-check that this merged correctly with the recent Haar measure change.

**Comment by Strilanc** [comment]
Yes! It worked! CirqBot merged into a fork!

**Pull request merged**